aslam_cv/aslam_cameras/include/aslam/cameras/implementation/CameraGeometry.hpp

```diff
@@ -2,756 +2,764 @@
+      return _projection.keypointToEuclidean(keypoint, outPoint)
+          && _mask.isValid(keypoint);
+    }
+
+    template<typename P, typename S, typename M>
+    template<typename DERIVED_K, typename DERIVED_P, typename DERIVED_JK>
+    bool CameraGeometry<P, S, M>::keypointToEuclidean(
+        const Eigen::MatrixBase<DERIVED_K> & keypoint,
+        const Eigen::MatrixBase<DERIVED_P> & outPoint,
+        const Eigen::MatrixBase<DERIVED_JK> & outJk) const {
+      // I'm putting the mask second in this case to make sure the keypointToEuclidean function is called.
+      return _projection.keypointToEuclidean(keypoint, outPoint, outJk)
+          && _mask.isValid(keypoint);
+    }
+
+    template<typename P, typename S, typename M>
+    template<typename DERIVED_K, typename DERIVED_P, typename DERIVED_JK>
+    bool CameraGeometry<P, S, M>::keypointToEuclideanFiniteDifference(
+        const Eigen::MatrixBase<DERIVED_K> & keypoint,
+        const Eigen::MatrixBase<DERIVED_P> & outPoint,
+        const Eigen::MatrixBase<DERIVED_JK> & outJk) const {
+      DERIVED_JK & Jk = const_cast<Eigen::MatrixBase<DERIVED_JK>&>(outJk).derived();
+      ASLAM_CAMERAS_ESTIMATE_JACOBIAN(this, keypointToEuclidean, keypoint, 1e-5,
+                                      Jk);
+
+      // I'm putting the mask second in this case to make sure the keypointToEuclidean function is called.
+      return _projection.keypointToEuclidean(keypoint, outPoint)
+          && _mask.isValid(keypoint);
+    }
+
+    template<typename P, typename S, typename M>
+    template<typename DERIVED_K, typename DERIVED_P>
+    bool CameraGeometry<P, S, M>::keypointToHomogeneous(
+        const Eigen::MatrixBase<DERIVED_K> & keypoint,
+        const Eigen::MatrixBase<DERIVED_P> & outPoint) const {
+      // I'm putting the mask second in this case to make sure the keypointToEuclidean function is called.
+      return _projection.keypointToHomogeneous(keypoint, outPoint)
+          && _mask.isValid(keypoint);
+    }
+
+    template<typename P, typename S, typename M>
+    template<typename DERIVED_K, typename DERIVED_P, typename DERIVED_JK>
+    bool CameraGeometry<P, S, M>::keypointToHomogeneous(
+        const Eigen::MatrixBase<DERIVED_K> & keypoint,
+        const Eigen::MatrixBase<DERIVED_P> & outPoint,
+        const Eigen::MatrixBase<DERIVED_JK> & outJk) const {
+      // I'm putting the mask second in this case to make sure the keypointToEuclidean function is called.
+      return _projection.keypointToHomogeneous(keypoint, outPoint, outJk)
+          && _mask.isValid(keypoint);
+    }
+
+    template<typename P, typename S, typename M>
+    template<typename DERIVED_K, typename DERIVED_P, typename DERIVED_JK>
+    bool CameraGeometry<P, S, M>::keypointToHomogeneousFiniteDifference(
+        const Eigen::MatrixBase<DERIVED_K> & keypoint,
+        const Eigen::MatrixBase<DERIVED_P> & outPoint,
```

Comment:
Just out of curiosity: why are these const, when they mark somewhat of a return value and the constness is cast away almost right away?